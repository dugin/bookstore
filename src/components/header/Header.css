.Header {
  justify-content: center;
  display: flex;
  align-items: center;
  min-height: 65px;
  box-shadow: 12px 15px 20px 0 rgba(46, 61, 73, 0.15);
  background-color: #3fc35f; }
  .Header__title {
    color: #fff; }
    .Header__title:hover {
      color: #fff; }
  .Header__links {
    position: absolute;
    right: 10px; }
    .Header__links__icon {
      font-size: 32px;
      color: #fff; }
      .Header__links__icon:hover {
        cursor: pointer; }
