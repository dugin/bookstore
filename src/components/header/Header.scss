@import '../../variables';

.Header {
  justify-content: center;
  display: flex;
  align-items: center;
  min-height: 65px;
  box-shadow: 12px 15px 20px 0 rgba(46, 61, 73, 0.15);
  background-color: $color__primary;

  &__title {
    color: $color__text--white;

    &:hover{
      color: $color__text--white;
    }
  }

  &__links {
    position: absolute;
    right: 10px;

    &__icon {
      font-size: 32px;
      color: $color__text--white;

      &:hover{
        cursor: pointer;
      }

    }
  }
}