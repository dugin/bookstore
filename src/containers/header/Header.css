.Header {
  box-shadow: 12px 15px 20px 0 rgba(46, 61, 73, 0.15);
  position: sticky;
  top: 0;
  z-index: 10;
  background-color: rgba(255, 255, 255, 0.85);
  border-bottom: 1px solid #d1d1d1; }
  .Header__block {
    display: flex;
    align-items: center;
    height: 60px; }
    .Header__block__back-arrow {
      position: relative;
      left: -20px;
      margin-top: 5px;
      font-size: 48px;
      color: #3fc35f; }
      @media (min-width: 768px) {
        .Header__block__back-arrow {
          position: absolute;
          left: 0;
          top: 0; } }
  .Header__title {
    color: #3fc35f;
    position: relative; }
    @media (min-width: 768px) {
      .Header__title {
        left: 50%;
        transform: translateX(-50%); } }
    .Header__title:hover {
      color: #3fc35f; }
  .Header__links {
    display: flex;
    flex-direction: row-reverse;
    flex-grow: 1;
    height: 100%; }
    .Header__links__wrapper {
      display: flex;
      align-items: center;
      padding-left: 0;
      height: 100%;
      border-left: 1px solid #fff; }
      .Header__links__wrapper__icon {
        padding-left: 15px;
        font-size: 32px;
        color: #3fc35f; }
        .Header__links__wrapper__icon:hover {
          cursor: pointer; }
      .Header__links__wrapper__priceContainer {
        color: #939393;
        display: flex;
        justify-content: center;
        flex-direction: column;
        margin-left: 5px; }
        .Header__links__wrapper__priceContainer__amount {
          margin-bottom: 3px;
          font-size: 12px;
          font-weight: bold; }
        .Header__links__wrapper__priceContainer__price {
          margin: 0;
          font-size: 12px;
          color: #3fc35f; }
