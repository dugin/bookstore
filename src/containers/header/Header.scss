@import '../../variables';

.Header {
  box-shadow: 12px 15px 20px 0 rgba(46, 61, 73, 0.15);
  position: sticky;
  top: 0;
  z-index: 10;
  background-color: rgba(255, 255, 255, .85);
  border-bottom: 1px solid $color__lineCard--gray;

  &__block {
    display: flex;
    align-items: center;
    height: 60px;

    &__back-arrow {
      position: relative;
      left: -20px;
      margin-top: 5px;
      font-size: 48px;
      color: $color--primary;

      @media (min-width: $screen-sm-min) {
        position: absolute;
        left: 0;
        top: 0;
      }
    }

  }

  &__title {
    color: $color--primary;
    position: relative;

    @media (min-width: $screen-sm-min) {
      left: 50%;
      transform: translateX(-50%);
    }

    &:hover {
      color: $color--primary;
    }
  }

  &__links {
    display: flex;
    flex-direction: row-reverse;
    flex-grow: 1;
    height: 100%;

    &__wrapper {
      display: flex;
      align-items: center;
      padding-left: 0;
      height: 100%;
      border-left: 1px solid $color--white;

      &__icon {
        padding-left: 15px;
        font-size: 32px;
        color: $color--primary;

        &:hover {
          cursor: pointer;
        }

      }

      &__priceContainer {
        color: $color__text--gray;
        display: flex;
        justify-content: center;
        flex-direction: column;
        margin-left: 5px;

        &__amount {
          margin-bottom: 3px;
          font-size: 12px;
          font-weight: bold;
        }
        &__price {
          margin: 0;
          font-size: 12px;
          color: $color--primary;
        }
      }

    }

  }
}