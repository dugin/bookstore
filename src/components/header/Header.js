import React from 'react';
import {Collapse, Navbar, NavbarToggler, NavbarBrand, Nav, NavItem, NavLink} from 'reactstrap';
import './Header.css'

const Header = (props) => {

    return (
        <header className="Header">
            <NavbarBrand className="Header__title" href="/">Bookstore</NavbarBrand>
            <Nav className="Header__links">
                <NavItem>
                    <NavLink><i className="Header__links__icon material-icons">shopping_cart</i></NavLink>
                </NavItem>
            </Nav>
        </header>
    )

};
export default Header;