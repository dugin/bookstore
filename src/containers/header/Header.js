import React, {Component} from 'react';
import './Header.css';
import {Link} from 'react-router-dom';
import {connect} from 'react-redux';
import NumberFormat from 'react-number-format';
import * as utils from '../../utils/booksUtils';
import {withRouter} from 'react-router-dom';

export class Header extends Component {

    constructor(props) {
        super(props);

        this.state = {route: window.location.pathname};

        this.listenToRouteChange();
    }

    listenToRouteChange() {
        this.props.history && this.props.history
            .listen((location) => {
                this.setState({route: location.pathname})
            });
    }

    setAmount = () => {
        return utils.setBooksAmount(this.props.books);
    };

    setPrice = () => {
        return utils.setPriceSum(this.props.books);
    };

    render() {
        return (
            <div className="Header">

                <nav className="container px-3 px-sm-0 Header__block">
                    {this.state.route && this.state.route.localeCompare('/') !== 0 && (
                        <a onClick={() => this.props.history.goBack()}>
                            <i className="material-icons Header__block__back-arrow">keyboard_arrow_left</i>
                        </a>
                    )}
                    <span className="Header__title navbar-brand" href="/">Livraria</span>
                    <div className="Header__links">

                        {this.state.route && this.state.route.localeCompare('/carrinho') !== 0 && (
                        <Link to="/carrinho" className="Header__links__wrapper">
                            <i className="Header__links__wrapper__icon material-icons">shopping_cart</i>

                            {this.props.books && this.props.books.length > 0 && (
                                <div className="Header__links__wrapper__priceContainer">
                                    <p className="Header__links__wrapper__priceContainer__amount">
                                        {this.setAmount()} livros</p>

                                    <NumberFormat className="Header__links__wrapper__priceContainer__price"
                                                  value={this.setPrice()}
                                                  decimalPrecision={2} displayType={'text'} decimalSeparator=','
                                                  prefix={'R$ '}/>
                                </div>
                            )}

                        </Link>
                        )}

                    </div>
                </nav>
            </div>
        );
    }
}

export default connect((state, props) =>({ books: state.books}), null)(withRouter(Header));