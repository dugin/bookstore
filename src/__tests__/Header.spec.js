import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Header, mapStateToProps} from '../containers/header/Header';
import {props} from "../__mocks__/bookMock";

describe('Header', () => {
    const header = shallow(<Header/>);

    it('renders properly', () => {
        expect(toJson(header)).toMatchSnapshot();
    });

    it('contains the title `Livraria` ', () => {
        expect(header.contains('Livraria')).toBe(true);
    });

    it('contains the cart icon', () => {
        expect(header.find('i').contains('shopping_cart')).toBe(true);
    });

    it('contains total value and amount of books in the cart when not empty', () => {
        header.setProps({books: [props]});

        const amount = header.find('.Header__links__wrapper__priceContainer__amount');
        const price = header.find('.Header__links__wrapper__priceContainer__price');

        expect(price.exists()).toBe(true);
        expect(amount.exists()).toBe(true);

        expect(header.instance().setPrice()).toEqual(80);
        expect(header.instance().setAmount()).toEqual(2);

    });

    it('mapStateToProps properly', () => {

        expect(mapStateToProps(props, null)).toEqual({book: header.instance().props.book});
    });

    describe('arrow left icon', () => {
        function setProps(props) {
            return shallow(<Header {...props} />);
        }

        it('show it when route is not home path', () => {
            header.setState({route: '/carrinho'});

            expect(header.find('.Header__block__back-arrow').contains('keyboard_arrow_left')).toBe(true);
        });

        it('does not show it when is home path', () => {
            header.setState({route: '/'});

            expect(header.find('.Header__block__back-arrow').contains('keyboard_arrow_left')).toBe(false);
        });

        it('click on it', () => {
            header.setState({route: '/carrinho'});
            header.setProps({history: {goBack: jest.fn()}});

            const backBtn = header.find('.Header__block__back-arrow').parent();

            backBtn.simulate('click');

            expect(header.instance().props.history.goBack).toHaveBeenCalled();
        });
    });

});