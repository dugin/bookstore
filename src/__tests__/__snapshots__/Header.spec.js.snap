// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders properly 1`] = `
<div
  className="Header"
>
  <nav
    className="container px-3 px-sm-0 Header__block"
  >
    <span
      className="Header__title navbar-brand"
      href="/"
    >
      Livraria
    </span>
    <div
      className="Header__links"
    >
      <Link
        className="Header__links__wrapper"
        replace={false}
        to="/carrinho"
      >
        <i
          className="Header__links__wrapper__icon material-icons"
        >
          shopping_cart
        </i>
      </Link>
    </div>
  </nav>
</div>
`;
